src/prng/mod.rs

```diff
@@ -8,37 +8,310 @@
+//! consideration is security. Performance and simplicity are also important,
+//! but in general CSPRNGs are more complex and slower than regular PRNGs.
+//! Quality is no longer a concern, as it is a requirement for a
+//! CSPRNG that the output is basically indistinguishable from true randomness
+//! since any bias or correlation makes the output more predictable.
+//!
+//! There is a close relationship between CSPRNGs and cryptographic ciphers.
+//! Any block cipher can be turned into a CSPRNG by encrypting a counter. Stream
+//! ciphers are basically a CSPRNG and a combining operation, usually XOR. This
+//! means that we can easily use any stream cipher as a CSPRNG.
+//!
+//! Rand currently provides two trustworthy CSPRNGs and two CSPRNG-like PRNGs:
+//!
+//! | name | full name |  performance | initialization | memory | predictability | forward secrecy |
+//! |------|-----------|--------------|--------------|----------|----------------|-------------------------|
+//! | [`ChaChaRng`] | ChaCha20 | ★☆☆☆☆ | fast | 136 bytes | secure | no |
+//! | [`Hc128Rng`] | HC-128 | ★★☆☆☆ | slow | 4176 bytes | secure | no |
+//! | [`IsaacRng`] | ISAAC | ★★☆☆☆ | slow | 2072 bytes | unknown | unknown |
+//! | [`Isaac64Rng`] | ISAAC-64 | ★★☆☆☆ | slow | 4136 bytes| unknown | unknown |
+//!
+//! It should be noted that the ISAAC generators are only included for
+//! historical reasons, they have been with the Rust language since the very
+//! beginning. They have good quality output and no attacks are known, but have
+//! received little attention from cryptography experts.
+//!
+//!
+//! # Performance
+//!
+//! First it has to be said most PRNGs are very fast, and will rarely be a
+//! performance bottleneck.
+//!
+//! Performance of basic PRNGs is a bit of a subtle thing. It depends a lot on
+//! the CPU architecture (32 vs. 64 bits), inlining, and also on the number of
+//! available registers. This often causes the performance to be affected by
+//! surrounding code due to inlining and other usage of registers.
+//!
+//! When choosing a PRNG for performance it is important to benchmark your own
+//! application due to interactions between PRNGs and surrounding code and
+//! dependence on the CPU architecture as well as the impact of the size of
+//! data requested. Because of all this, we do not include performance numbers
+//! here but merely a qualitative rating (which is not comparable between PRNGs
+//! and CSPRNGs).
+//!
+//! CSPRNGs are a little different in that they typically generate a block of
+//! output in a cache, and pull outputs from the cache. This allows them to have
+//! good amortised performance, and reduces or completely removes the influence
+//! of surrounding code on the CSPRNG performance.
+//!
+//! ### Worst-case performance
+//! Because CSPRNGs usually produce a block of values into a cache, they have
+//! poor worst case performance (in contrast to regular PRNGs).
```

Comment:
> regular PRNGs, which are usually constant-time

Reply:
I don't really want to use the term "constant-time" here, as it has special meaning in cryptography. But I'll try something.